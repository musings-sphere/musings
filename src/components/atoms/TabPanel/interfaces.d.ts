export interface TabPanelProps {
	children: JSX.Element;
	value: string | string[] | number | null;
	index: string | string[] | number | null;
}
