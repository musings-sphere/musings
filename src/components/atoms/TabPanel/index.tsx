import { Box } from '@mui/material';
// interfaces
import { TabPanelProps } from '@components/atoms/TabPanel/interfaces';

const TabPanel = ({
	children,
	value,
	index,
	...other
}: TabPanelProps): JSX.Element => {
	return (
		<div
			role="tabpanel"
			hidden={value !== index}
			id={`menu-tabpanel-${index}`}
			aria-labelledby={`menu-tab-${index}`}
			{...other}
		>
			{value === index && (
				<Box
					sx={{
						padding: 0,
						margin: 0,
					}}
					p={3}
					{...other}
					className="tab-panel"
					data-testid="tab-panel"
				>
					{children}
				</Box>
			)}
		</div>
	);
};

export default TabPanel;
