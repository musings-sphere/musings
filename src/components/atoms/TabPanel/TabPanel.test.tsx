// react libraries
import { render, screen } from '@testing-library/react';

// component
import TabPanel from './index';

describe('TabPanel component', () => {
	const props = {
		index: 1,
		value: 1,
		children: <div />,
	};

	const { asFragment } = render(<TabPanel {...props} />);

	it('should render correctly', () => {
		expect(asFragment()).toMatchSnapshot();

		const elem = screen.getByTestId('tab-panel');
		expect(elem).toHaveClass('tab-panel');
	});
});
