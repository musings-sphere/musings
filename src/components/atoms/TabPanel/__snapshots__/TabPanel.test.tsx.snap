// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabPanel component should render correctly 1`] = `
<DocumentFragment>
  <div
    aria-labelledby="menu-tab-1"
    id="menu-tabpanel-1"
    role="tabpanel"
  >
    <div
      class="tab-panel MuiBox-root css-cu9xnh"
      data-testid="tab-panel"
    >
      <div />
    </div>
  </div>
</DocumentFragment>
`;
